File: src/main/kotlin/com/gondev/todolist/model/Survey.kt
package com.gondev.todolist.model

import com.gondev.todolist.model.audit.OwnerShipAudit
import org.hibernate.annotations.Fetch
import org.hibernate.annotations.FetchMode
import javax.persistence.CascadeType
import javax.persistence.Entity
import javax.persistence.FetchType
import javax.persistence.OneToMany

@Entity
data class Survey(
        var title: String="",

        var description: String?=null,

        @OneToMany(mappedBy = "survey", cascade = [CascadeType.ALL], fetch = FetchType.LAZY, orphanRemoval = true)
        @Fetch(FetchMode.SELECT)
        var questions:MutableList<Question> = arrayListOf()
):OwnerShipAudit()

File: frontend-vue.ts/src/response/survey.ts
import {User, UserImp} from "@/response/user";
import {Question} from "@/response/question";

export interface Survey {
	title: string;
	description?: string;
	questions?: Question[];
	id: number;
	createAt: string;
	modifyAt: string;
	user: User
}

export class SurveyImp implements Survey {
	public title: string = "";
	public description?: string;
	public id: number = 0;
	public createAt: string = "";
	public modifyAt: string = "";
	public user: User = new UserImp();
}
